ctapipe/io/eventfilereader.py

```diff
@@ -148,216 +130,90 @@ def check_file_compatibility(file_path):
         """
 
     @property
-    @abstractmethod
-    def num_events(self):
+    def metadata(self):
```

Comment:
why is `metadata` a property again and not just a simple attribute? 
Usually one would make it a property when one wants to do more than only returning it.